src/main/java/by/itacademy/javaenterprise/kotkovski/dao/impl/CarRepositoryImpl.java

```diff
@@ -0,0 +1,98 @@
+package by.itacademy.javaenterprise.kotkovski.dao.impl;
+
+import by.itacademy.javaenterprise.kotkovski.dao.CarRepository;
+import by.itacademy.javaenterprise.kotkovski.entity.Car;
+import by.itacademy.javaenterprise.kotkovski.exception.DAOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class CarRepositoryImpl implements CarRepository {
+
+    private final static Logger logger = LoggerFactory.getLogger(CarRepositoryImpl.class);
+
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    private final CustomerRepository customerRepository = new CustomerRepository();
```

Comment:
```suggestion
    private CustomerRepository customerRepository;
```

Это грубая ошибка. 
С одной стороны ты хочешь использовать депендеси инжекчин, с другой стороны ты прям тут создаешь объект.

- [x] использовать dependency injection